bee-common/bee-packable/src/packable/bounded.rs

```diff
@@ -132,35 +171,55 @@ macro_rules! bounded {
     };
 }
 
-bounded!(BoundedU8, InvalidBoundedU8, u8);
-bounded!(BoundedU16, InvalidBoundedU16, u16);
-bounded!(BoundedU32, InvalidBoundedU32, u32);
-bounded!(BoundedU64, InvalidBoundedU64, u64);
+bounded!(BoundedU8, InvalidBoundedU8, TryIntoBoundedU8Error, u8);
+bounded!(BoundedU16, InvalidBoundedU16, TryIntoBoundedU16Error, u16);
+bounded!(BoundedU32, InvalidBoundedU32, TryIntoBoundedU32Error, u32);
+bounded!(BoundedU64, InvalidBoundedU64, TryIntoBoundedU64Error, u64);
 
 impl Bounded for u8 {
     type Bounds = Self;
+    type Range = Range<Self>;
 
     const MIN: Self::Bounds = u8::MIN;
     const MAX: Self::Bounds = u8::MAX;
+
+    fn range(start: Self::Bounds, end: Self::Bounds) -> Self::Range {
+        start..end
+    }
```

Comment:
Shouldn't this be a default implementation ?

Reply:
I don't see any reason why it shouldn't. I'll try to change it and see what happens.